src/NServiceBus.Core/Routing/MessageEndpointMappingCollection.cs

```diff
@@ -120,7 +120,14 @@ public void Add(MessageEndpointMapping mapping)
         /// </summary>
         protected override void BaseAdd(ConfigurationElement element)
         {
-            BaseAdd(element, true);
+            try
+            {
+                BaseAdd(element, true);
+            }
+            catch (ConfigurationErrorsException e)
+            {
+                throw new Exception($"An ambigious message endpoint mapping has been defined at line: {e.Line}. Check the 'MessageEndpointMappings' section in {e.Filename} for conflicting mappings.", e);
```

Comment:
Perhaps we could get then endpoint property out using https://msdn.microsoft.com/en-us/library/system.configuration.configurationelement.properties(v=vs.110).aspx and display that?

`ambigious message endpoint mapping has been defined for {endpoint}` ?